ext/cache/01_cache.js

```diff
@@ -141,16 +141,18 @@
         },
       );
       if (reader) {
+        let success = false;
         try {
           while (true) {
             const { value, done } = await reader.read();
             if (done) {
+              success = true;
               break;
             }
             await core.write(rid, value);
           }
         } finally {
-          await core.shutdown(rid);
+          await core.opAsync("op_shutdown_with_status", rid, success);
```

Comment:
The missed in the previous PR. We should call shutdown() only if done is true. And keep core.close(rid) in the finally block.